TrafficCapture/trafficReplayer/src/test/java/org/opensearch/migrations/replay/SigV4SignerTest.java

```diff
@@ -0,0 +1,217 @@
+package org.opensearch.migrations.replay;
+
+import io.netty.buffer.Unpooled;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.net.ssl.SSLException;
+import lombok.SneakyThrows;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.util.Base64;
+import java.nio.charset.Charset;
+import io.netty.buffer.ByteBuf;
+import org.opensearch.migrations.replay.datahandlers.http.HttpJsonMessageWithFaultingPayload;
+import org.opensearch.migrations.testutils.SimpleHttpResponse;
+import org.opensearch.migrations.testutils.SimpleHttpClientForTesting;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import java.util.Map;
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class SigV4SignerTest {
```

Comment:
The strategy that I would use here would be to manually confirm that a specific use case works with a SigV4 service.  Validate that the service accepts the signature.  Then make a similar request with exactly the same parameters (header values, timestamp, etc).  Confirm that you can generate the same signature with the same credentials (say with DefaultCredentialsProvider wired to pull your default profile).  Once you can confirm that the signatures match, now set the credentials to a dummy set.  Do the full calculation, notice that the newly signed request has different signature values.  Replace the old non-matched values with those from the new mocked credentials provider and voila, now you can test that the signing logic is valid.

These tests don't work as unit tests since they depend upon a network connection and a specific set of credentials, which won't be available to every user.